src/monstergenerator.cpp

```diff
@@ -532,11 +563,24 @@ void mtype::load( JsonObject &jo )
 
     const typed_flag_reader<decltype( gen.flag_map )> flag_reader{ gen.flag_map, "invalid monster flag" };
     optional( jo, was_loaded, "flags", flags, flag_reader );
+    // Can't calculate yet - we want all flags first
+    optional( jo, was_loaded, "bash_skill", bash_skill, -1 );
 
     const typed_flag_reader<decltype( gen.trigger_map )> trigger_reader{ gen.trigger_map, "invalid monster trigger" };
     optional( jo, was_loaded, "anger_triggers", anger, trigger_reader );
     optional( jo, was_loaded, "placate_triggers", placate, trigger_reader );
     optional( jo, was_loaded, "fear_triggers", fear, trigger_reader );
+
+    if( jo.has_member( "path_settings" ) ) {
+        auto jop = jo.get_object( "path_settings" );
+        // Here rather than in pathfinding.cpp because we want monster-specific defaults and was_loaded
+        optional( jop, was_loaded, "max_dist", path_settings.max_dist, 0 );
+        optional( jop, was_loaded, "max_length", path_settings.max_length, -1 );
+        optional( jop, was_loaded, "bash_strength", path_settings.bash_strength, -1 );
+        optional( jop, was_loaded, "allow_open_doors", path_settings.allow_open_doors, false );
```

Comment:
NPCs use the same structure and they both open doors an avoid traps.

Setting `allow_open_doors: true` won't actually make monsters open doors, it will only make them treat doors as openable during pathfinding.